monkey/tests/unit_tests/monkey_island/cc/agent_event_handlers/test_scan_event_handler.py

```diff
@@ -189,16 +162,19 @@ def handler(scan_event_handler, request):
 
 
 def test_ping_scan_event_target_machine_not_exists(
-    scan_event_handler, machine_repository: IMachineRepository
+    in_memory_machine_repository: IMachineRepository, agent_repository, node_repository
```

Comment:
Ahh, it makes more sense in the `test_node_update_facade`

Reply:
It simplifies a number of unit tests. Instead of needing to create a MagicMock() and then define some boundaries, some tests just need a `MachineRepository` that will behave according to the interface.